themes-default/slim/views/vue-components/app-link.mako

```diff
@@ -22,80 +23,109 @@ Vue.component('app-link', {
-                                return location.href + newHash;
-                            }
-                            return location.href.replace(location.hash, '') + href;
-                        }
-                        if (isIRC(href)) {
-                            return href;
-                        }
-                        if (isAbsolute(href)) {
-                            if (isExternal(href)) {
-                                return anonymise(href)
-                            } else {
-                                return href;
-                            }
-                        } else {
-                            return new URL(href, base).href
-                        }
-                    };
-                    return {
-                        is: 'a',
-                        target: isAbsolute(href) && isExternal(href) ? '_blank' : '_self',
-                        href: resolvedHref(),
-                        rel: isAbsolute(href) && isExternal(href) ? 'noreferrer' : undefined
-                    };
-                }
 
-                // Just return a boring link with other attrs
-                // @NOTE: This is for scroll achors as it uses the id
+            // Just return a boring link with other attrs
+            // @NOTE: This is for scroll achors as it uses the id
+            if (!href) {
                 return {
                     is: 'a',
                     falseLink: true
                 };
             }
+
+            // @TODO: Remove this once we move to vue only
+            if (isAnonymised) {
+                throw new Error('Still using anon_url in Python!');
+            }
+
             return {
-                is: 'router-link',
-                to: { name: href }
+                is: 'a',
+                target: isAbsolute && isExternal ? '_blank' : '_self',
+                href: (() => {
+                    if (isHashPath) {
+                        const { location } = window;
+                        if (location.hash.length === 0) {
+                            // Current location might be `url#`
+                            const newHash = location.href.endsWith('#') ? href.substr(1) : href;
+                            return location.href + newHash;
+                        }
+                        return location.href.replace(location.hash, '') + href;
+                    }
+                    if (isIRC) {
+                        return href;
+                    }
+                    if (isAbsolute) {
+                        if (isExternal) {
+                            return anonymisedHref;
+                        }
+                        return href;
+                    }
+                    return new URL(href, base).href;
+                })(),
+                rel: isAbsolute && isExternal ? 'noreferrer' : undefined
             };
         }
     },
-    template: `#app-link-template`
+    template: '#app-link-template'
```

Comment:
I think our style is to put the template property right below the component name. Let's do that :)

Reply:
We need to create a styles guide for things like this on the wiki.